src/main.rs

```diff
@@ -17,7 +17,7 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
         )
         .get_matches();
 
-    let market_url = cli.value_of("marketUrl").expect("No market URL provided!");
+    let market_url = cli.value_of("marketUrl").expect("No market URL provided!").to_string();
```

Comment:
Good job at using `expect()`, since we do `require` the `marketUrl` this should be there (this is why we use `expect()`) if anything goes wrong this will panic (crashing the program)